File: Matrike/Matrix.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class Matrix {
	static final int DIM = 500;
	
	public static void readNumbers(String filename, long[][] arr) {
        try (BufferedReader br = new BufferedReader(new FileReader(filename))) {
            for (int i = 0; i < DIM; i++) {
                for (int j = 0; j < DIM; j++) {
					String line = br.readLine();
                    arr[i][j] = Long.parseLong(line);
                }
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
	
    public static void mnozenjeMatrik(long[][] A, long[][] C) {        
        for (int i = 0; i < DIM; i++) {
            for (int j = 0; j < DIM; j++) {
                for (int k = 0; k < DIM; k++) {
                    C[i][j] += A[i][k] * A[k][j];
                }
            }
		}
    }

    public static void main(String[] args) {
        long[][] A = new long[DIM][DIM];
		long[][] C = new long[DIM][DIM];
		readNumbers("numbers.txt",A);
        long startTime = System.nanoTime();
		mnozenjeMatrik(A, C);
        long endTime = System.nanoTime();
		double timeTaken = (endTime-startTime) / 1E6;
		System.out.println("Time taken: "+ timeTaken +" milliseconds");
		System.out.println("Number: "+ C[0][0] );
    }
}

File: Matrike/matrix.py
import time

DIM = 500

def readNumbers(file,matrix):
    with open("numbers.txt", "r") as file:
        for row in range(500):
            for col in range(500):
                line = file.readline().strip()
                try:
                    num = int(line)
                    matrix[row][col] = num
                except ValueError:
                    col-=1


        
def mnozenjeMatrik(A, C):
    for i in range(DIM):
        for j in range(DIM):
            for k in range(DIM):
                C[i][j] += A[i][k] * A[k][j]



A = [[0 for _ in range(DIM)] for _ in range(DIM)]
C = [[0 for _ in range(DIM)] for _ in range(DIM)]
readNumbers("numbers.txt",A) 
startTime = time.process_time()
mnozenjeMatrik(A, C)
endTime = time.process_time()
timeTaken = (endTime - startTime) * 1000.0
print(f"Execution time: {timeTaken:.3f} milliseconds")
print(f"Number: ",C[0][0])

